### ISSUE TITLE
Add 60ghz Wireless stuff

### ISSUE THREAD

**Issue by durandguru** [terraform-routeros/terraform-provider-routeros#618]
**Is your feature request related to a problem? Please describe.**
Add the 60ghz wireless stuff

**Describe the solution you'd like**
Add the commands for:
AP + Client -> /interface w60g
AP -> /interface w60g station

**Documentation**
https://help.mikrotik.com/docs/spaces/ROS/pages/39059501/W60G

The main interface under "/interface w60g" is always there, an default interface. On AP and client the same.

On the AP clients can be predefined under "/interface w60g station" but also can be created by routeros as an client connect with the correct settings

**Additional context**
I of couse can test it. And Merry Christmas!

**Comment by vaerh** [comment]
Ok, I'll do it later. Do you have json from this resource? 

Merry Christmas to you, too!

**Comment by durandguru** [comment]
/interface w60g ->

[
   {
      ".id":"*2",
      "arp":"enabled",
      "arp-timeout":"auto",
      "beamforming-event":"1858",
      "default-scan-list":"58320,60480,62640,64800",
      "disabled":"false",
      "frequency":"60480",
      "isolate-stations":"false",
      "l2mtu":"1600",
      "mac-address":"48:8F:5A:F3:34:E0",
      "mode":"ap-bridge",
      "mtu":"1500",
      "name":"wlan60-1",
      "password":"password",
      "put-stations-in-bridge":"bridge",
      "region":"eu",
      "running":"false",
      "rx-mpdu-crc-err":"20354232",
      "rx-mpdu-crc-ok":"1401642596",
      "rx-ppdu":"1403448812",
      "ssid":"ptmp01",
      "tx-fw-msdu":"52",
      "tx-io-msdu":"0",
      "tx-mpdu-new":"202533008",
      "tx-mpdu-retry":"3570458",
      "tx-mpdu-total":"206103466",
      "tx-ppdu":"3230579961",
      "tx-ppdu-from-q":"2031518609",
      "tx-sector":"auto",
      "tx-sw-msdu":"202533128"
   }
]

![image](https://github.com/user-attachments/assets/84a4b83c-b75e-459b-8490-75eebf05e2bd)

![image](https://github.com/user-attachments/assets/230c9699-133c-4224-a6dc-4b386c4c8f24)

![image](https://github.com/user-attachments/assets/91a5bb19-0ebd-4c9d-b89d-3a39362e36a6)

**Comment by durandguru** [comment]
/interface w60g station -> 

[
   {
      ".id":"*8",
      "arp":"enabled",
      "arp-timeout":"auto",
      "beamforming-event":"3042",
      "disabled":"false",
      "mac-address":"48:8F:5A:F3:34:E0",
      "mtu":"1500",
      "name":"ptp01",
      "parent":"wlan60-1",
      "put-in-bridge":"parent",
      "remote-address":"48:8F:5A:47:81:9E",
      "running":"true"
   }
]

![image](https://github.com/user-attachments/assets/ed5aaa2c-d2a4-47aa-99d3-9bd83d30838f)

![image](https://github.com/user-attachments/assets/aa07e783-9126-47c2-8c78-008a8f387352)

**Comment by durandguru** [comment]
And take your time!

**Comment by durandguru** [comment]
Thanks for adding this! Only had now time to test stuff. The following remarks:
-the routeros_interface_w60g manages an system created interface, So it now tries to create and fails. Importing does work and then i can manage the resource ;-)
-The field region is not an bool. but you select something from an list

![Image](https://github.com/user-attachments/assets/c8d0923f-8cf6-482d-8a06-a99807995f6d)

-Missing field:
 Warning: Field 'default_scan_list' not found in the schema
│
│   with module.test[0].routeros_interface_w60g.ap[0],
│   on test\main.tf line 1, in resource "routeros_interface_w60g" "ap":
│    1: resource "routeros_interface_w60g" "ap" {
│
│ [MikrotikResourceDataToTerraform] The field was lost during the Schema development: ▷ 'default_scan_list': '58320,60480,62640,64800' ◁

For routeros_interface_w60g_station, the field mac_address is required for creating the resource.

**Comment by durandguru** [comment]
@vaerh can not reopen, or would you like me to create a new issue?

**Comment by vaerh** [comment]
I suggest we stay here.

**Comment by vaerh** [comment]
Quite a few misses.))
I'll make the fixes in the next release and then its request to test again. Same as always)

**Comment by vaerh** [comment]
Hi, @durandguru !

>-the routeros_interface_w60g manages an system created interface

Does MT provide for the creation of virtual interfaces via this menu or does it always control the `w60` system interface?

**Comment by durandguru** [comment]
@vaerh did you see my last message? No rush but wonderd if you have seen it.

**Comment by vaerh** [comment]
I apologize, I missed it in the mailbox.

**Comment by vaerh** [comment]
@durandguru 
Hi!

I'm trying to figure out what the problem might be and I realize I don't have enough data to do that. Can I have a look (without private data) at a single resource config and a failed exchange with MT?

**Comment by durandguru** [comment]
Sorry, took me some time to find time for this.

resource "routeros_interface_w60g" "test" {
  name     = "wlan60-1"
  password = "put_your_safe_password_here"
  ssid     = "put_your_new_ssid_here"
  disabled = false
  mode     = "ap-bridge"
}

Error: POST 'https://172.31.10.31:443/rest/interface/w60g/set' returned response code: 400, message: 'Bad Request', details: 'missing =.id='
│
│   with routeros_interface_w60g.test,
│   on main.tf line 40, in resource "routeros_interface_w60g" "test":
│   40: resource "routeros_interface_w60g" "test" {

routeros_interface_w60g.test: Creating...
2025-05-13T09:49:00.552+0200 [INFO]  Starting apply for routeros_interface_w60g.test
2025-05-13T09:49:00.552+0200 [DEBUG] routeros_interface_w60g.test: applying the planned Create change
2025-05-13T09:49:00.553+0200 [DEBUG] provider.terraform-provider-routeros_v1.84.0.exe: request body:  {"disabled":"no","mode":"ap-bridge","name":"wlan60-1","password":"put_your_safe_password_here","ssid":"put_your_new_ssid_here"}: @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=872a2946-0b48-683a-17cf-2a2bcd372c11 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 timestamp="2025-05-13T09:49:00.553+0200"
2025-05-13T09:49:00.553+0200 [DEBUG] provider.terraform-provider-routeros_v1.84.0.exe: POST request URL:  https://172.31.10.31:443/rest/interface/w60g/set: tf_provider_addr=terraform-routeros/routeros tf_req_id=872a2946-0b48-683a-17cf-2a2bcd372c11 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros timestamp="2025-05-13T09:49:00.553+0200"
2025-05-13T09:49:00.564+0200 [DEBUG] provider.terraform-provider-routeros_v1.84.0.exe: error response body:
{"detail":"missing =.id=","error":400,"message":"Bad Request"}: tf_req_id=872a2946-0b48-683a-17cf-2a2bcd372c11 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros timestamp="2025-05-13T09:49:00.564+0200"
2025-05-13T09:49:00.564+0200 [ERROR] provider.terraform-provider-routeros_v1.84.0.exe: Response contains error diagnostic: diagnostic_severity=ERROR tf_proto_version=5.8 tf_provider_addr=terraform-routeros/routeros tf_req_id=d18881d8-8f9c-8529-61cd-70640ef1f4f9 tf_resource_type=routeros_interface_w60g @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail="" diagnostic_summary="POST 'https://172.31.10.31:443/rest/interface/w60g/set' returned response code: 400, message: 'Bad Request', details: 'missing =.id='" tf_rpc=ApplyResourceChange timestamp="2025-05-13T09:49:00.564+0200"
2025-05-13T09:49:00.565+0200 [DEBUG] State storage *remote.State declined to persist a state snapshot
2025-05-13T09:49:00.565+0200 [ERROR] vertex "routeros_interface_w60g.test" error: POST 'https://172.31.10.31:443/rest/interface/w60g/set' returned response code: 400, message: 'Bad Request', details: 'missing =.id='

**Comment by vaerh** [comment]
@durandguru 
Do I remember correctly that w60g are system interfaces that cannot be deleted, but we can create a new virtual w60g interface?

**Comment by durandguru** [comment]
Yeah the W60G is an system interface. On that you can't add or delete an interface.  (In 60g station we can add/remove, that one is working fine)

![Image](https://github.com/user-attachments/assets/5c5a2bb1-1e11-4a94-9dcc-4b31dac6ff36)
